main.js

```diff
@@ -386,9 +392,22 @@ $(function() {
       showMessage('0 units in town - can not buy.');
       return;
     }
-    inventory[productIndex] += 1;
-    towns[currentTown].stockpiles[productIndex] -= 1;
-    money -= price;
+
+    //if player selected "buy all", count is set to highest purchesable amount
+    const all = towns[currentTown].stockpiles[productIndex];
+    const buttonText = $(this).text(); 
+    const count = ( buttonText != "all") ? 1
```

Comment:
Please use the strict comparison operator `!==`. (It's a good practice to not rely on or even allow type coercion, with the debatable exception of null-checks.)